src/network/C4Network2IO.cpp

```diff
@@ -1264,15 +1292,23 @@ void C4Network2IO::SendConnPackets()
 
 void C4Network2IO::OnPuncherConnect(C4NetIO::addr_t addr)
 {
-	// Sanity check
-	if (addr.GetFamily() != C4NetIO::HostAddress::IPv4)
-		return;
-	Application.InteractiveThread.ThreadLogS("Adding address from puncher: %s", addr.ToString().getData());
+	// NAT punching is only relevant for IPv4, so convert here to show a proper address.
```

Comment:
It does connect back and the same code will be called. It just won't detect any new addresses, provided there's no IPv6 NAT. AddAddr checks for duplicates, so it's okay.

We could skip a bit of code, but I don't think it's worth the extra check here. This only happens once per game anyways.